File: MakingCoolUIs/app/src/main/java/com/example/makingcooluis/MainActivity2.java
package com.example.makingcooluis;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

public class MainActivity2 extends AppCompatActivity {

    CardView googleCard, facebookCard;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        googleCard=findViewById(R.id.googleCard);
        facebookCard=findViewById(R.id.facebookCard5);

        googleCard.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(getApplicationContext(),"You clicked google",Toast.LENGTH_SHORT).show();
                Intent intent= new Intent(getApplicationContext(),MainActivity3.class);
                startActivity(intent);
            }
        });

        facebookCard.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(getApplicationContext(),"You clicked facebook",Toast.LENGTH_SHORT).show();
            }
        });

    }
}

File: FirstKotlin/app/src/main/java/com/example/firstkotlin/activities/MainActivity2.kt
package com.example.firstkotlin.activities

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.firstkotlin.R
import kotlinx.android.synthetic.main.activity_main2.*

class MainActivity2 : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        //main activity 2
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)

        val someText : String?= intent.extras?.getString("text2")
        text.setText(someText)
    }
}